## 01_MediaItem.diff by Andrew Lake <jamboarder@yahoo.com>
## Fix a FTBFS with gcc-4.4
diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index dbccde7..6523c2c 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -834,7 +834,7 @@ void MainWindow::delayedNotificationHide()
     QTimer::singleShot(3000, ui->notificationWidget, SLOT(hide()));
 }
 
-void MainWindow::sourceInfoUpdated(MediaItem mediaItem)
+void MainWindow::sourceInfoUpdated(const MediaItem &mediaItem)
 {
     QFontMetrics fm =  ui->notificationText->fontMetrics();
 	//TODO: I tried fixing this word puzzle, I couldn't test it myself, but it should work. Please check if it really is :-) thanks
diff --git a/src/mainwindow.h b/src/mainwindow.h
index c456831..bb546bd 100644
--- a/src/mainwindow.h
+++ b/src/mainwindow.h
@@ -39,7 +39,7 @@
 #include <QDateTime>
 #include <QMainWindow>
 
-class MediaItem;
+struct MediaItem;
 class MediaListProperties;
 class MediaItemModel;
 class MediaListCache;
@@ -189,7 +189,7 @@ private slots:
     void showLoading();
     void showNotification();
     void delayedNotificationHide();
-    void sourceInfoUpdated(MediaItem mediaItem);
+    void sourceInfoUpdated(const MediaItem &mediaItem);
     void sourceInfoRemoved(QString url);
     void updateNowPlayingStyleSheet();
     void volumeChanged(qreal newVolume);
